### PR TITLE
Migrate 0.7.2 to cosmos-sdk goz-phase-3

### PR DESCRIPTION
**Pull request by alpe**
See #128 

### Reviewer note
The `--validate-signatures` is not supported anymore in `cli_test.go` so I replaced failing tests with the matching ones from https://github.com/cosmos/cosmos-sdk/blob/goz-phase-3/x/auth/client/cli/cli_test.go
______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: app/codec.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package app
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/std"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func MakeCodecs() (*std.Codec, *codec.Codec) {
```
Comment (ethanfrey) [inline app/codec.go:10]: This is getting rather complex. Every iteration of the sdk seems to make it harder to use

File: cli_test/test_helpers.go [hunk 2]
```diff
@@ -127,12 +125,11 @@ func (f Fixtures) GenesisFile() string {
 
 // GenesisFile returns the application's genesis state
 func (f Fixtures) GenesisState() simapp.GenesisState {
-	cdc := codec.New()
 	genDoc, err := tmtypes.GenesisDocFromFile(f.GenesisFile())
 	require.NoError(f.T, err)
 
 	var appState simapp.GenesisState
-	require.NoError(f.T, cdc.UnmarshalJSON(genDoc.AppState, &appState))
+	require.NoError(f.T, f.cdc.UnmarshalJSON(genDoc.AppState, &appState))
```
Comment (ethanfrey) [inline cli_test/test_helpers.go:132]: Good check here

File: app/app.go [hunk 3]
```diff
@@ -262,12 +257,12 @@ func NewWasmApp(
 
 	// Create IBC Keeper
 	app.ibcKeeper = ibc.NewKeeper(
-		app.cdc, keys[ibc.StoreKey], app.stakingKeeper, scopedIBCKeeper,
+		app.cdc, appCodec, keys[ibc.StoreKey], app.stakingKeeper, scopedIBCKeeper,
```
Comment (ethanfrey) [inline app/app.go:260]: I am getting confused.

Why does this need both `*std.Codec` and `*codec.Codec` and what are these for?

(Not your issue, but if you can give me a 1 paragraph issue, I would be grateful)

File: x/wasm/keeper/test_common.go [hunk 4]
```diff
@@ -56,7 +56,9 @@ func CreateTestInput(t *testing.T, tempDir string) (sdk.Context, auth.AccountKee
 	isCheckTx := false
 	ctx := sdk.NewContext(ms, abci.Header{}, isCheckTx, log.NewNopLogger())
 	cdc := MakeTestCodec()
-	appCodec := codecstd.NewAppCodec(cdc)
+	interfaceRegistry := codectypes.NewInterfaceRegistry()
```
Comment (ethanfrey) [inline x/wasm/keeper/test_common.go:59]: Should we just pull in the `MakeCodec` from `app` to this file and use it?

It seems it did a number of register commands that might be needed to some infrastructure later. I did spend ~2 hours hunting down a bug that was caused by not registering some amino codec in my test code.

File: app/app.go (hunk 3 above)
Comment (alpe) [reply app/app.go:260]: I have not looked into details but migration to protobuf is not done for this module

File: x/wasm/keeper/test_common.go (hunk 4 above)
Comment (alpe) [reply x/wasm/keeper/test_common.go:59]: good idea but this creates cyclic dependencies:
```
package github.com/cosmwasm/wasmd/app
	imports github.com/cosmwasm/wasmd/x/wasm
	imports github.com/cosmwasm/wasmd/x/wasm/client/cli
	imports github.com/cosmwasm/wasmd/x/wasm/keeper
	imports github.com/cosmwasm/wasmd/app
```

Comment (ethanfrey) [reply x/wasm/keeper/test_common.go:59]: I actually meant cut-and-paste that code :grimacing: 

Too many circular deps in the sdk. Anyway, good to merge as is, or with this cut and paste.

**Comment by ethanfrey** [comment]
One note... this seems to close #128 not #127 

(you are updating the sdk, not merging v0.8 into this)

**Pull request closed without merging**